src/main/java/appeng/fluids/container/ContainerFluidLevelEmitter.java

```diff
@@ -0,0 +1,97 @@
+
+package appeng.fluids.container;
+
+
+import net.minecraft.client.gui.GuiTextField;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.InventoryPlayer;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import appeng.api.config.RedstoneMode;
+import appeng.api.config.SecurityPermissions;
+import appeng.api.config.Settings;
+import appeng.container.guisync.GuiSync;
+import appeng.fluids.parts.PartFluidLevelEmitter;
+import appeng.fluids.util.IAEFluidTank;
+import appeng.util.Platform;
+
+
+public class ContainerFluidLevelEmitter extends ContainerFluidConfigurable
+{
+	private final PartFluidLevelEmitter lvlEmitter;
+
+	@SideOnly( Side.CLIENT )
+	private GuiTextField textField;
+	@GuiSync( 3 )
+	public long EmitterValue = -1;
+
+	public ContainerFluidLevelEmitter( final InventoryPlayer ip, final PartFluidLevelEmitter te )
+	{
+		super( ip, te );
+		this.lvlEmitter = te;
+	}
+
+	@SideOnly( Side.CLIENT )
+	public void setTextField( final GuiTextField level )
+	{
+		this.textField = level;
+		this.textField.setText( String.valueOf( this.EmitterValue ) );
+	}
+
+	public void setLevel( final long l, final EntityPlayer player )
+	{
+		this.lvlEmitter.setReportingValue( l );
+		this.EmitterValue = l;
+	}
+
+	@Override
+	protected void setupConfig()
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1186)